relays/ethereum/src/message_race_delivery.rs

```diff
@@ -229,18 +266,19 @@ impl<P: MessageLane> RaceStrategy<SourceHeaderIdOf<P>, TargetHeaderIdOf<P>, P::M
 		}
 
 		// 1) we want to deliver all nonces, starting from `target_nonce + 1`
-		// 2) we want to deliver at most `MAX_MESSAGES_TO_RELAY_IN_SINGLE_TX` nonces in this batch
+		// 2) we want to deliver at most `self.max_nonces_to_relay_in_single_tx` nonces in this batch
 		// 3) we can't deliver new nonce until header, that has emitted this nonce, is finalized
 		// by target client
 		let nonces_begin = self.target_nonce + 1.into();
 		let best_header_at_target = &race_state.target_state.as_ref()?.best_peer;
 		let mut nonces_end = None;
-		for i in 0..MAX_MESSAGES_TO_RELAY_IN_SINGLE_TX {
-			let nonce = nonces_begin + i.into();
+		let mut i = Zero::zero();
+		while i < self.max_nonces_to_relay_in_single_tx {
+			let nonce = nonces_begin + i;
```

Comment:
```suggestion
		let mut nonce = nonces_begin;
		let mut end = nonces_begin + self.max_nonces_to_relay_in_single_tx;
		while nonce < end {
```

Wouldn't that be a bit simpler?